Automattic/wp-calypso
# FIX | Jetpack sites with standalone plugins are mistakenly recognised as WPCOM sites in Calypso

#### Some background

We have a predicate/selector in this repo named `isJetpackSite` which (before this PR) returned true if a site had a full Jetpack plugin because it expected a boolean `jetpack` flag on the site object. Now, since we now have Standalone Jetpack products, that boolean flag remains `false` if a user installs only a standalone plugin.  Reason being mentioned here p1628888267118900/1628888111.118000-slack-C025MP1Q3HT.
Now, when that happened, we began noticing problems in calypso when users install standalone plugins.

I tried to fix such an issue in the past 
- #60527

Here are more PRs that tried to solve the same problem:
- #56005
- #55506
- #62955
- #56276
- #63641

Those PRs (including mine) duplicated the logic and worked only as patches for a bigger problem.

Here is another issue that was closed and re-opened a few times because the same problem re-surfaced elsewhere.

- #56194

The whole problem began with #56005 which, instead of enhancing `isJetpackSite` (to consider sites with standalone Jetpack products as Jetpack sites), created `isJetpackProductSite`. So, everywhere else Calypso continued treating sites with only standalone Jetpack products as Non-Jetpack sites. Many such issues were fixed in the above-listed PRs, but no effort was made to actually fix the problem at its root.

#55506 introduced some changes (with no given reason in the PR) to remove sites with standalone plugins from the site selector. It created problems for teams managing standalone plugins. Then in #62955 Jetpack React team tried to fix the problem introduced by those changes. The fact is that #55506 was not needed at all because we don't want to hide the sites in the site selector if they have a standalone plugin installed. Thus, this PR reverts both of those changes.

This PR attempts to fix the issue at its root and thus it changes the default behavior of `isJetpackSite`.

Asana task: 1164141197617539-as-1202810120032857/f

#### Proposed Changes

* Enhance `isJetpackSite` to consider standalone products when deciding whether a site is a Jetpack site or not.
* Add unit tests for `isJetpackSite`
* Remove `isJetpackProductSite` selector which actually resulted in more confusion and problems than it solved.
* Remove `hasJetpackPluginActiveConnection` for the same reasons
* Update the fallback `JETPACK_ICON` to use the correct icon in Calypso, because standalone plugins can't allow WPCOM to get the correct icons from the site in the absence of JSON API (Provided by full Jetpack plugin).

#### Testing Instructions

- Now create a JN site without Jetpack plugin or uninstall Jetpack after the site is ready
- Goto wp-admin > Plugins > Add New
- Search for Jetpack Backup which is the standalone backup plugin
- Activate and connect Jetpack with Backup plan.
- Boot this PR and run `yarn start`
- Goto `http://calypso.localhost:3000/backup/:site`
- Confirm that you see the backup UI
- Try to switch the site
- Confirm that the site is listed in the site selector
- Run `yarn start-jetpack-cloud-p` and goto [http://jetpack.cloud.localhost:3001](http://jetpack.cloud.localhost:3001)
- Confirm that you see the site in site selector
- Select the site
- Confirm that you see the corresponding plan UI i.e. Backup
- Confirm that upsells work correctly
- Goto a UI which is meant for WPCOM sites. e.g. `http://calypso.localhost:3000/domains/manage/:site`
- Confirm that you see "Domains are not available for this site."
- Create a JN site with full Jetpack plugin
- Connect Jetpack and purchase a plan like Backup or Security
- Confirm the above steps for the site with this standalone plugin.

Also, please test as per the instructions in these related PRs:
- #60527
- #56005
- #62955
- #63641
- #56276

#### Pre-merge Checklist





These screenshots are after installing the standalone backup plugin

| BEFORE | AFTER |
| - | - |
|<img width="1142" alt="Screenshot 2022-08-17 at 5 19 23 PM" src="https://user-images.githubusercontent.com/18226415/185114256-c0f10509-7398-49ca-8441-9359d5201e02.png">|<img width="1141" alt="Screenshot 2022-08-17 at 5 20 04 PM" src="https://user-images.githubusercontent.com/18226415/185114271-16e69ad6-e09b-437e-9379-7cf489f3bce4.png">|
|<img width="1679" alt="Screenshot 2022-08-17 at 5 22 29 PM" src="https://user-images.githubusercontent.com/18226415/185114120-798e0277-1fe9-473f-8f88-fbf6768b0e70.png">|<img width="1681" alt="Screenshot 2022-08-17 at 5 23 34 PM" src="https://user-images.githubusercontent.com/18226415/185114138-9513e099-2fb8-4300-9651-efa88f5f1f99.png">|
|<img width="1673" alt="Screenshot 2022-08-17 at 5 24 46 PM" src="https://user-images.githubusercontent.com/18226415/185113932-459f0c45-d65e-418a-acc3-0350cb2146ba.png">|<img width="1667" alt="Screenshot 2022-08-17 at 5 25 29 PM" src="https://user-images.githubusercontent.com/18226415/185113945-a9c9329a-c12e-47a5-8eec-b3ea4fdab285.png">|
|<img width="1682" alt="Screenshot 2022-08-17 at 6 22 23 PM" src="https://user-images.githubusercontent.com/18226415/185136636-d9dc0da1-e7f5-44e4-bbcf-5328dd9d9d3c.png">|<img width="1683" alt="Screenshot 2022-08-17 at 6 23 08 PM" src="https://user-images.githubusercontent.com/18226415/185136701-5e4ac6a7-7437-4c83-9cdb-1e3e8e1df109.png">|

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202810120032857
